src/brownian_diffusive_flux.cc

```diff
@@ -37,17 +37,17 @@ static double calculateFlux(int idx,
     if (!no_flux)
         {
         // average density at left edge
-        auto rho_avg = 0.5*(rho(idx-1)+rho(idx));
+        auto rho_avg = mesh->linearInterpolate(idx,rho);
```

Comment:
This would look like:
```suggestion
        auto rho_avg = mesh->interpolate(idx,rho);
```